### PR TITLE
Features/2fa

### PR DESCRIPTION
**Pull request by FlorianM11**

### CODE REVIEW

File: src/EventSuscriber/DoubleAuthentificationSuscriber.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    public function onKernelRequest(RequestEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $route = $event->getRequest()->attributes->get('_route');
+        if (\in_array($route, ['app_login', 'app_security_validate_authentification', 'homepage', 'images', 'app_security_authentification'], true)) {
```
Comment (guillaume-sainthillier) [inline src/EventSuscriber/DoubleAuthentificationSuscriber.php:53]: Could you change this by a blacklist with the `app_security_authentification_protected` route?

File: src/Controller/SecurityController.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            'error' => $error,
+        ]);
+    }
+
+    /**
+     * @Route("/2FA-protected", name="app_security_authentification_protected")
+     */
+    public function authentificationProtected()
+    {
+        if ($this->getUser()) {
```
Comment (guillaume-sainthillier) [inline src/Controller/SecurityController.php:101]: This statement will be useless if you change the Suscriber (see below)

File: templates/security/login.html.twig [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+{% block body %}
+    <form method="post">
+        {% if error %}
+            <div class="alert alert-danger">
+                <div class="container px-0">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
+            </div>
+        {% endif %}
+
+        <div class="container mt-5 mb-5">
+            <h1 class="h3 mb-3 font-weight-normal">Connectez-vous</h1>
```
Comment (guillaume-sainthillier) [inline templates/security/login.html.twig:14]: Please use english labels

File: templates/security/valide_authentification.html.twig [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 
                 <div class="form-group">
-                    <label for="qrCode">Code de validation</label>
+                    <label for="qrCode">Validate code</label>
                     <input type="text" id="qrCode" name="qrCode" class="form-control" required="required" />
                 </div>
                 <input type="submit" class="btn btn-primary" value="Confirmer" />
```
Comment (guillaume-sainthillier) [inline templates/security/valide_authentification.html.twig:18]: I think you forget some french translations :-)

Comment (FlorianM11) [reply templates/security/valide_authentification.html.twig:18]: shit

**Review by guillaume-sainthillier — COMMENTED** [review]
Could you please rename `connexion` (french) to `connection` (english)?

**Comment by guillaume-sainthillier** [comment]
Same for menu link

**Pull request merged**